checker/tests/optional/Shadowing.java

```diff
@@ -0,0 +1,34 @@
+import java.util.Optional;
+import org.checkerframework.checker.optional.qual.*;
+import org.checkerframework.dataflow.qual.*;
+
+@SuppressWarnings({"optional:field", "optional:parameter"})
+public class Shadowing {
+
+  public Optional<String> f;
+
+  public Optional<String> g;
+
+  class Sub extends Shadowing {
+    Optional<String> f;
+
+    void foo(@Present Optional<String> present) {
+      super.f = present;
+      @Present Optional<String> ok = super.f;
+    }
+
+    void bar(@Present Optional<String> present) {
+      this.f = present;
+      @Present Optional<String> ok = this.f;
+    }
+
+    void baz(@Present Optional<String> present) {
+      super.g = present;
+      @Present Optional<String> ok1 = this.g;
```

Comment:
This is an error in this pull request but not master.

Reply:
If the store now correctly normalizes the expression `super.g` to `this.g` to have the value `@Present` here, I would expect `this.g` to also be `@Present` in the RHS of the assignment, unless it has no information about the supertype, might that be the case?

Reply:
There is no normalization from `super.g` to `this.g`.  That's the problem.

Reply:
I see.

Would it make sense to do checks here: https://github.com/typetools/checker-framework/blob/0232f9daf47a6ff850c322ffe3e2ce12ffaa890e/dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java#L536-L538
 
Since the “this” in this case might actually be referring to a supertype, I think there are also a number of other locations where
the same (or similar) logic might apply.

> On May 30, 2024, at 8:44 AM, Suzanne Millstein ***

Reply:
I've fixed this test case. ([d964c4d](https://github.com/typetools/checker-framework/pull/6611/commits/d964c4da11e64cec1931d53754ea3f9fa85669ad)) The same change probably needs to be made for method calls.

Reply:
Thanks so much! I'll take a look at making that change.

Reply:
Suzanne, I've added [this test class](https://github.com/typetools/checker-framework/pull/6611/commits/9654e7e7a001a9a5368aa89bdb9c1b30e1e98ed1) for method calls involving `super` and `this`.

It appears to pass; does this imply that we actually don't need to make the same change for method calls?